src/devices/Bmxx80/Bmxx80Base.cs

```diff
@@ -0,0 +1,207 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.IO;
+using Iot.Device.Bmxx80.CalibrationData;
+using Iot.Device.Bmxx80.PowerMode;
+using Iot.Device.Bmxx80.Register;
+using Iot.Units;
+
+namespace Iot.Device.Bmxx80
+{
+    /// <summary>
+    /// Represents the core functionality of the Bmxx80 family.
+    /// </summary>
+    public abstract class Bmxx80Base : IDisposable
+    {
+        internal Bmxx80CalibrationData _calibrationData;
+        internal I2cDevice _i2cDevice;
+        internal CommunicationProtocol _communicationProtocol;
+        internal byte _controlRegister;
+
+        internal enum CommunicationProtocol
+        {
+            I2c
+        }
+
+        /// <summary>
+        /// The variable _temperatureFine carries a fine resolution temperature value over to the
+        /// pressure compensation formula and could be implemented as a global variable.
+        /// </summary>
+        protected int TemperatureFine;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Bmxx80Base"/> class.
+        /// </summary>
+        /// <param name="deviceId">The ID of the device.</param>
+        /// <param name="i2cDevice">The <see cref="I2cDevice"/> to create with.</param>
+        /// <exception cref="ArgumentNullException"></exception>
+        /// <exception cref="IOException"></exception>
```

Comment:
NIT: While we are adding it, do you mind also adding a brief text here explaining when are the exceptions thrown?